### ISSUE TITLE
segmentation fault, Ubuntu16.04. Cuda=9.2, cudnn=7.1, GPU=1 CUDNN=1  OPENCV=1

### ISSUE THREAD

**Issue by TimeFunny** [pjreddie/darknet#919]
OPENCV : 3.3

- [ ] GPU=1, OPENCV =1       **OK**
- [ ] GPU=1    **OK**
- [ ] 
- [ ] but  GPU=1, CUDNN=1, OPENCV=1 

```
> ./darknet detector train cfg/voc.data cfg/yolov3-voc.cfg darknet53.conv.74
..........
Loading weights from darknet53.conv.74...Done!
Learning Rate: 0.001, Momentum: 0.9, Decay: 0.0005
Loaded: 0.793186 seconds
Segmentation fault (core dumped)
> gdb ./darknet ./core 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Core was generated by `./darknet detector train cfg/voc.data cfg/yolov3-voc.cfg darknet53.conv.74'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007f8e18a56d0d in cudnnBatchNormalizationForwardTraining () from /usr/local/cuda/lib64/libcudnn.so.7
[Current thread is 1 (Thread 0x7f8e3acc5a40 (LWP 42992))]
(gdb) 
(gdb) where
#0  0x00007f8e18a56d0d in cudnnBatchNormalizationForwardTraining () from /usr/local/cuda/lib64/libcudnn.so.7
#1  0x00000000004b8b4c in forward_batchnorm_layer_gpu (l=..., net=...) at ./src/batchnorm_layer.c:197
#2  0x00000000004cc0e3 in forward_convolutional_layer_gpu (l=..., net=...) at ./src/convolutional_kernels.cu:127
#3  0x0000000000485776 in forward_network_gpu (netp=0x1de4970) at ./src/network.c:778
#4  0x0000000000482afa in forward_network (netp=0x1de4970) at ./src/network.c:192
#5  0x00000000004831e5 in train_network_datum (net=0x1de4970) at ./src/network.c:293
#6  0x00000000004833bf in train_network (net=0x1de4970, d=...) at ./src/network.c:324
#7  0x0000000000432685 in train_detector (datacfg=0x7ffd23a29651 "cfg/voc.data", cfgfile=0x7ffd23a2965e "cfg/yolov3-voc.cfg", 
    weightfile=0x7ffd23a29671 "darknet53.conv.74", gpus=0x7ffd23a27874, ngpus=1, clear=0) at ./examples/detector.c:118
#8  0x000000000043625b in run_detector (argc=6, argv=0x7ffd23a27a68) at ./examples/detector.c:842
#9  0x000000000043cbe9 in main (argc=6, argv=0x7ffd23a27a68) at ./examples/darknet.c:440
(gdb) 
#0  0x00007f8e18a56d0d in cudnnBatchNormalizationForwardTraining () from /usr/local/cuda/lib64/libcudnn.so.7
#1  0x00000000004b8b4c in forward_batchnorm_layer_gpu (l=..., net=...) at ./src/batchnorm_layer.c:197
#2  0x00000000004cc0e3 in forward_convolutional_layer_gpu (l=..., net=...) at ./src/convolutional_kernels.cu:127
#3  0x0000000000485776 in forward_network_gpu (netp=0x1de4970) at ./src/network.c:778
#4  0x0000000000482afa in forward_network (netp=0x1de4970) at ./src/network.c:192
#5  0x00000000004831e5 in train_network_datum (net=0x1de4970) at ./src/network.c:293
#6  0x00000000004833bf in train_network (net=0x1de4970, d=...) at ./src/network.c:324
#7  0x0000000000432685 in train_detector (datacfg=0x7ffd23a29651 "cfg/voc.data", cfgfile=0x7ffd23a2965e "cfg/yolov3-voc.cfg", 
    weightfile=0x7ffd23a29671 "darknet53.conv.74", gpus=0x7ffd23a27874, ngpus=1, clear=0) at ./examples/detector.c:118
#8  0x000000000043625b in run_detector (argc=6, argv=0x7ffd23a27a68) at ./examples/detector.c:842
#9  0x000000000043cbe9 in main (argc=6, argv=0x7ffd23a27a68) at ./examples/darknet.c:440
(gdb) bt
#0  0x00007f8e18a56d0d in cudnnBatchNormalizationForwardTraining ()
   from /usr/local/cuda/lib64/libcudnn.so.7
#1  0x00000000004b8b4c in forward_batchnorm_layer_gpu (l=..., net=...)
    at ./src/batchnorm_layer.c:197
#2  0x00000000004cc0e3 in forward_convolutional_layer_gpu (l=..., 
    net=...) at ./src/convolutional_kernels.cu:127
#3  0x0000000000485776 in forward_network_gpu (netp=0x1de4970)
    at ./src/network.c:778
#4  0x0000000000482afa in forward_network (netp=0x1de4970)
    at ./src/network.c:192
#5  0x00000000004831e5 in train_network_datum (net=0x1de4970)
    at ./src/network.c:293
#6  0x00000000004833bf in train_network (net=0x1de4970, d=...)
    at ./src/network.c:324
#7  0x0000000000432685 in train_detector (
    datacfg=0x7ffd23a29651 "cfg/voc.data", 
    cfgfile=0x7ffd23a2965e "cfg/yolov3-voc.cfg", 
    weightfile=0x7ffd23a29671 "darknet53.conv.74", gpus=0x7ffd23a27874, 
    ngpus=1, clear=0) at ./examples/detector.c:118
#8  0x000000000043625b in run_detector (argc=6, argv=0x7ffd23a27a68)
    at ./examples/detector.c:842
#9  0x000000000043cbe9 in main (argc=6, argv=0x7ffd23a27a68)
    at ./examples/darknet.c:440
(gdb) 
#0  0x00007f8e18a56d0d in cudnnBatchNormalizationForwardTraining ()
   from /usr/local/cuda/lib64/libcudnn.so.7
#1  0x00000000004b8b4c in forward_batchnorm_layer_gpu (l=..., net=...)
    at ./src/batchnorm_layer.c:197
#2  0x00000000004cc0e3 in forward_convolutional_layer_gpu (l=..., 
    net=...) at ./src/convolutional_kernels.cu:127
#3  0x0000000000485776 in forward_network_gpu (netp=0x1de4970)
    at ./src/network.c:778
#4  0x0000000000482afa in forward_network (netp=0x1de4970)
    at ./src/network.c:192
#5  0x00000000004831e5 in train_network_datum (net=0x1de4970)
    at ./src/network.c:293
#6  0x00000000004833bf in train_network (net=0x1de4970, d=...)
    at ./src/network.c:324
#7  0x0000000000432685 in train_detector (
    datacfg=0x7ffd23a29651 "cfg/voc.data", 
    cfgfile=0x7ffd23a2965e "cfg/yolov3-voc.cfg", 
    weightfile=0x7ffd23a29671 "darknet53.conv.74", gpus=0x7ffd23a27874, 
    ngpus=1, clear=0) at ./examples/detector.c:118
#8  0x000000000043625b in run_detector (argc=6, argv=0x7ffd23a27a68)
    at ./examples/detector.c:842
#9  0x000000000043cbe9 in main (argc=6, argv=0x7ffd23a27a68)
    at ./examples/darknet.c:440
```
 the segmentation fault is caused by the function cudnnBatchNormalizationForwardTraining() in cudnn. But i hava no idea to solve it

**Comment by coolbei** [comment]
check your cfg file carefully.  I ran into the same problem, it turned out to be a typo in cfg file.

**Comment by TimeFunny** [comment]
Can  tell me  more infomation? I skip the cuda batchNorm function,  then it run.

**Comment by kecsap** [comment]
@TimeFunny Update you cudnn to the latest and check it again. I had crash with CUDA 9.2+cudnn 7.0 with custom built Tensorflow yesterday. After I updated cudnn and rebuilt everything, the crash went away.
